libusb/descriptor.c

```diff
@@ -59,9 +59,14 @@ static void parse_descriptor(const void *source, const char *descriptor, void *d
 			sp += 2;
 			dp += 2;
 			break;
-		case 'd':	/* 32-bit word, convert from little endian to CPU */
+		case 'd':	/* 32-bit word, convert from little endian to CPU (4-byte align dst before write). */
 			dp += 4 - ((uintptr_t)dp & 3);	/* Align to 32-bit word boundary */
 
+			*((uint32_t *)dp) = READ_LE32(sp);
+			sp += 4;
+			dp += 4;
+			break;
+		case 'i':	/* 32-bit word, convert from little endian to CPU (no dst alignment before write) */
 			*((uint32_t *)dp) = READ_LE32(sp);
 			sp += 4;
 			dp += 4;
```

Comment:
This function (before your patch!) was pretty dubious, so I created https://github.com/libusb/libusb/pull/1460

Reply:
How does it work, should I wait for your PR to get merged and rebase or the other way around?

Reply:
Either way is fine with me. Your change here is fine, and perfectly in-line with the current style of that file/function. Just seeing that crazy pointer manipulation when reviewing this motivated me to create the other patch.